### PR TITLE
feat: move to typescript code micro-generator

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**

### CODE REVIEW

**Comment by xiaozhenliu-gg5** [comment]
Hi Alex, please review this PR, and docs-test fail because there are three broken links in docs/, I see the way to replace the broken links in synth.py, but I'm wondering where do I get the correct link and properly replace them? Thanks!

**Comment by alexander-fenster** [comment]
The three failing doc links:

```
[404] http://localhost:5506/compute/docs/zones
[404] http://localhost:5506/compute/docs/networks-and-firewalls
[404] http://localhost:5506/container-engine/reference/rest/v1/projects.zones.clusters
```

should point to the corresponding URLs in cloud.google.com site. We normally fix those links in `synth.py` like here:

https://github.com/googleapis/nodejs-datacatalog/blob/4614c4be0b09802ab39e77acc3cad0435586177e/synth.py#L48

and also create a tracking issue like this one https://github.com/googleapis/nodejs-datacatalog/issues/62.

File: src/v1/cluster_manager_client_config.json [hunk 1]
```diff
@@ -36,143 +36,142 @@
...
         "DeleteCluster": {
-          "timeout_millis": 60000,
-          "retry_codes_name": "non_idempotent",
+          "timeout_millis": 20000,
+          "retry_codes_name": "idempotent",
           "retry_params_name": "default"
         },
         "ListOperations": {
-          "timeout_millis": 60000,
+          "timeout_millis": 20000,
```
Comment (alexander-fenster) [inline src/v1/cluster_manager_client_config.json]: Please double check that this timeout change is expected (the gRPC service config might be incorrect). If it is incorrect, we'll need to fix it separately.

**Comment by xiaozhenliu-gg5** [comment]
Thanks for the suggestions! checking the timeout value and repointing the doc link :)

**Comment by xiaozhenliu-gg5** [comment]
Timeout change is intentional, it's set to `20s` in github `googleapis/grpc_config`, and also `//google3` and `third_party`

**Comment by xiaozhenliu-gg5** [comment]
One question is that, broken links are in `docs`, we can replace them in `synthtool.py`, but when `npm run docs-test` runs, it will trigger `npm run docs` one more time and `predocs-test` will replace back all the broken links. So I run `synthtool` again before the docs-test.

**Comment by alexander-fenster** [comment]
LGTM now!

**Comment by xiaozhenliu-gg5** [comment]
Thanks for the review!

File: synth.py [hunk 2]
```diff
@@ -6,30 +6,49 @@
...
-        r"[resource quota](https://cloud.google.com/compute/docs/resource-quotas)")
+# fix broken doc links
 
 s.replace("src/v1/doc/google/container/v1/doc_cluster_service.js",
         "https:\/\/cloud\.google\.com\/kubernetes-engine\/docs\/reference\/rest\/v1\/projects\.zones\.clusters\.nodePool",
         "https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.zones.clusters.nodePools#resource-nodepool")
 
+s.replace('src/v1/*.ts',
+        '/compute/docs/zones#available',
+        'https://cloud.google.com/compute/docs/regions-zones/')
```
Comment (alexander-fenster) [inline synth.py]: the `#available` is lost here, do they have this anchor?

Comment (xiaozhenliu-gg5) [reply synth.py]: rechecked the ts file, it only contains `/compute/docs/zones` one replace should work:
```
s.replace('src/v1/*.ts',
        '/compute/docs/zones',
        'https://cloud.google.com/compute/docs/regions-zones/')
```

**Pull request merged**